### PR TITLE
fix(pipeline): treat child processes as heartbeat liveness signal

### LINKED ISSUE #917: retro: Gates stage killed by pipeline STALL_TIMEOUT at 420s while a legitimat
## Retro Finding R-01

Gates stage killed by pipeline STALL_TIMEOUT at 420s while a legitimately-running .NET test job was in progress in a background Bash (`dotnet test PPDS.sln --filter "Category!=Integration"`, 10-min agent timeout). All TypeScript gates had already passed. The pipeline heartbeat monitor uses stdout-byte advancement to detect activity, but the agent was using the Monitor tool to await the background process completion, which emits no transcript output while waiting. Result: monitor sees `output_bytes` static for 5 min â†’ STALL_TIMEOUT â†’ gates exit=-1.

**Recommended fix:** Make the pipeline stall detector aware of active background Bash jobs â€” either (a) treat `run_in_background=true` jobs as liveness signal while they remain running, (b) raise the stall threshold for the gates stage (dotnet test on PPDS.sln routinely exceeds 5 min on cold caches), or (c) revise the gates skill to avoid `run_in_background` on the longest-pole command (.NET tests) so Bash stdout trickle keeps the heartbeat warm.

---
*Filed automatically by pipeline retro.*

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary
- Add child process detection as a 4th heartbeat liveness signal in `classify_activity()`, preventing the pipeline stall timeout from killing gates while `dotnet test` runs legitimately in background
- Add `get_child_process_count(pid)` helper with cross-platform support (wmic on Windows, pgrep on Unix) and safe fallback to 0 on any error
- Filter wmic output to numeric PIDs only, avoiding false positives from "No Instance(s) Available." text

Closes #917

## Test Plan
- [x] 8 new tests (AC-147): has_children active signal, idle without children, stall prevention across beats, child process detection on Unix/Windows, timeout handling, wmic no-instances edge case
- [x] All 17 existing stall/heartbeat/activity tests still pass (backward compatible via keyword-only `has_children=False` default)
- [x] Full gate suite green: .NET build + 13,256 tests, TS build + typecheck + lint + 438 tests, TUI snapshots

## Verification
- [x] /gates passed
- [x] /verify completed (surfaces: workflow)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/pipeline.py [hunk 1]
```diff
@@ -269,8 +269,36 @@ def release_lock(lock_path):
...
+            result = subprocess.run(
+                ["wmic", "process", "where", f"ParentProcessId={pid}",
+                 "get", "ProcessId"],
+                capture_output=True, text=True, timeout=5,
+                encoding="utf-8", errors="replace",
+            )
+            if result.returncode == 0:
+                lines = [l.strip() for l in result.stdout.strip().splitlines()
+                         if l.strip().isdigit()]
+                return len(lines)
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/pipeline.py:285]: Not applicable — Suggested replacement is invalid: tasklist does not support a PARENTPROCESS filter (only STATUS/IMAGENAME/PID/SESSION/SESSIONNAME/CPUTIME/MEMUSAGE/USERNAME/SERVICES/WINDOWTITLE/MODULES per `tasklist /?`). The UTF-16LE encoding claim is contradicted by the recent fix b7f3a1938, which detected readable text (`'No Instance(s) Available.'` and `'ProcessId'` header) â€” wmic output decodes as text in practice. Return code IS checked (`if result.returncode == 0`); the broad exception catch is intentional fail-open for a liveness probe with multiple redundant signals (output growth, git changes, commits, children). wmic deprecation on newer Windows is a fair long-term concern but the fail-open `FileNotFoundError` path handles it gracefully.
